### PR TITLE
Loss

### PR DESCRIPTION
**Pull request by rousseab**
In this PR:
 - implementation of lightning module
 - loss computation in lightning module
 - creation of noisy positions 
 - clean up of how "noise" is generated.
 - some misc clean ups / improvements in the tests.

I also fixed a bug where I was sometimes computing $\sigma^2 S$, and sometimes computing $\sigma  S$, but calling it the same thing.

Finally, I implement a "sanity check" experiment where I try to overfit a small dataset of random positions with a MLP score network.  Despite fiddling around with parameters, you can see below that this model doesn't learn *anything*.

Screenshot 2024-03-18 at 4 42 28 PM

This is suspicious and will require further exploration. I don't know why this is happening, but my best guesses are 
1- the data in fact keeps changing because we are sampling noise randomly: the model doesn't see the same batch over and over
2- an MLP is really crappy.

### CODE REVIEW

File: crystal_diffusion/models/optimizer.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    Args:
+        hyper_params : hyperparameters defining the optimizer
+        model : A neural network model.
+
+    Returns:
+        optimizer : The optimizer for the given model
+    """
+    match hyper_params.name:
+        case ValidOptimizerNames.adam:
+            optimizer = optim.Adam(model.parameters(), lr=hyper_params.learning_rate)
```
Comment (sblackburn86) [inline crystal_diffusion/models/optimizer.py:36]: depending on the regime, we might have to change the eps variable in adam. It might be premature to add it at this stage, it is easy enough to do later. Just something to keep in mind.

**Review by sblackburn86 — APPROVED** [review]
